multiplatform-movies/common/src/commonMain/kotlin/com.novoda.movies.gallery/GalleryPresenter.kt

```diff
@@ -0,0 +1,37 @@
+package com.novoda.movies.gallery
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+class GalleryPresenter internal constructor(private val uiContext: CoroutineContext, private val galleryFetcher: GalleryFetcher) : CoroutineScope {
+
+    private lateinit var job: Job
+
+    override val coroutineContext: CoroutineContext
+        get() = uiContext + job
+
+
+    fun startPresenting(view: View) {
+        job = Job()
+
+        launch {
+            try {
+                val gallery = galleryFetcher.fetchGallery()
+                view.render(gallery)
+            } catch (exception: Exception) {
```

Comment:
Did you had a look at https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-exception-handler/index.html to handle Exceptions in coroutines?

Reply:
nice one 👍 
10fa827c